flora-pm/flora-server
# [NO-ISSUE] fix nix

## Proposed changes

- update nix code 
  - remove unnecessary custom overrides 
  - refactoring (restructuring, separate out large nix files) 
  - include linting and formatting 
  - unbreak the build (tests, haddock, run) 
  - focus on simplicity (please note in your review if the nix code is in some place 
    hard to read, such that I can add some comments
- new nix workflow 
  - check nix code 
  - check build in nix 
  - check formatting of nix and haskell code in nix ci 
- overhaul of the documentation
  - include docs on how to build `flora` with nix 
  - how to use `nix` to provision `devShell`s
- update the dependencies of `flora` 

> **Warning**
> action has to be taken by a maintainer to add a cachix api key 

## Contributor checklist

- ~~My PR is related to \<insert ticket number>~~ `[NO-ISSUE]`
- [x] I have read and understood the [CONTRIBUTING guide](https://github.com/flora-pm/flora-server/blob/development/CONTRIBUTING.md)
- [x] I have inserted my change and a link to this PR in the [CHANGELOG](https://github.com/flora-pm/flora-server/blob/development/CHANGELOG.md)
